rewrite-groovy/src/main/java/org/openrewrite/groovy/GroovyParserVisitor.java

```diff
@@ -77,6 +79,7 @@ public class GroovyParserVisitor {
     private final FileAttributes fileAttributes;
 
     private final String source;
+    private final Map<Integer, String> sourceByLineNumbers;
```

Comment:
Any reason not to use a plain ArrayList here?

Reply:
Further I would probably just have stored the offsets, rather than the actual lines as strings.